composer/callbacks/speed_monitor.py

```diff
@@ -95,6 +102,31 @@ def load_state_dict(self, state: Dict[str, Any]) -> None:
         )
         self.total_eval_wct = state['total_eval_wct']
 
+        # Added in 0.13. Load only if present to support backwards compatibility
+        if 'eval_wct_per_label' in state:
+            self.eval_wct_per_label = state['eval_wct_per_label']
+        if 'eval_rate_per_label' in state:
+            self.eval_rate_per_label = state['eval_rate_per_label']
+        if 'last_elapsed_duration' in state:
+            self.last_elapsed_duration = state['last_elapsed_duration']
+
+    def get_elapsed_duration(self, state: State) -> Optional[float]:
+        """Get the elapsed duration.
+
+        Unlike `state.get_elapsed_duration`, this method computes fractional progress in an epoch
+        provided at least 1 epoch has passed by recording how many batches were in each epoch.
+        """
+        if state.max_duration is None:
+            return None
+        elif state.max_duration.unit == TimeUnit('ep') and state.timestamp.epoch.value > 1:
+            batches_per_epoch = (state.timestamp.batch -
+                                 state.timestamp.batch_in_epoch).value / state.timestamp.epoch.value
+            return state.timestamp.get('ba').value / (state.max_duration.value * batches_per_epoch)
+        elapsed_dur = state.get_elapsed_duration()
+        if elapsed_dur:
+            return elapsed_dur.value
+        return None
```

Comment:
The last `return None` is because I unpack `elapsed_dur` with `.value`, so I need to None check it. I think in the rest of the codebase we don't use `else:` case if it ends in a return, rather prefering:
```
if ...
  return ...
return ...
```
so I think it makes sense to keep it like this to be consistent? wdyt?

Reply:
ah ok that's fine.  I guess then we can remove the elif then?